H2WithQuotingTest$SelectUseLiterals test fails

JDK 8 update 20

When I attempt to build https://github.com/cowwoc/querydsl/commit/2ff8cc6490fcd44c44259d1361334c9d77b637c9 I get the following failures:

```
  H2SuiteTest$SelectUseLiterals>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
  H2SuiteTest$Select>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
  H2WithQuotingTest$SelectUseLiterals>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
  H2WithQuotingTest$Select>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
  H2WithSchemaTest$SelectUseLiterals>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
  H2WithSchemaTest$Select>SelectBase.Date_Diff2:448 expected:<361> but was:<362>
```

Here is a sample stack-trace:

```
Date_Diff2(com.mysema.query.suites.H2WithSchemaTest$Select)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError: expected:<361> but was:<362>
    at org.junit.Assert.fail(Assert.java:91)
    at org.junit.Assert.failNotEquals(Assert.java:645)
    at org.junit.Assert.assertEquals(Assert.java:126)
    at org.junit.Assert.assertEquals(Assert.java:470)
    at org.junit.Assert.assertEquals(Assert.java:454)
    at com.mysema.query.SelectBase.Date_Diff2(SelectBase.java:448)
```

--- a/querydsl-sql/src/test/java/com/mysema/query/Constants.java
+++ b/querydsl-sql/src/test/java/com/mysema/query/Constants.java
@@ -17,6 +17,7 @@ import java.util.Calendar;
 
 import com.mysema.query.sql.domain.QEmployee;
 import com.mysema.query.sql.domain.QSurvey;
+import org.joda.time.LocalDate;
 
 public final class Constants {
 
@@ -35,14 +36,10 @@ public final class Constants {
     public static final QSurvey survey2 = new QSurvey("s2");
 
     static{
-        Calendar cal = Calendar.getInstance();
-        cal.clear();
-        cal.set(Calendar.YEAR, 2000);
-        cal.set(Calendar.MONTH, 1);
-        cal.set(Calendar.DAY_OF_MONTH, 2);
-        date = new java.sql.Date(cal.getTimeInMillis());
+        LocalDate localDate = new LocalDate(2000, 2, 10);
+        date = new java.sql.Date(localDate.toDateMidnight().getMillis());
 
-        cal = Calendar.getInstance();
+        Calendar cal = Calendar.getInstance();
         cal.set(1970, 0, 1, 3, 4);
         cal.set(Calendar.SECOND, 30);
         cal.set(Calendar.MILLISECOND, 0);

--- a/querydsl-sql/src/test/java/com/mysema/query/SelectBase.java
+++ b/querydsl-sql/src/test/java/com/mysema/query/SelectBase.java
@@ -430,8 +430,9 @@ public class SelectBase extends AbstractBaseTest {
     @ExcludeIn({CUBRID, DERBY, HSQLDB, SQLITE, TERADATA})
     public void Date_Diff2() {
         TestQuery query = query().from(employee).orderBy(employee.id.asc());
-        int offset = TimeZone.getDefault().getOffset(0);
-        Date date = new java.sql.Date(-offset);
+
+        LocalDate localDate = new LocalDate(1970, 1, 10);
+        Date date = new Date(localDate.toDateMidnight().getMillis());
 
         int years = query.singleResult(SQLExpressions.datediff(DatePart.year, date, employee.datefield));
         int months = query.singleResult(SQLExpressions.datediff(DatePart.month, date, employee.datefield));
@@ -441,10 +442,10 @@ public class SelectBase extends AbstractBaseTest {
         int minutes = query.singleResult(SQLExpressions.datediff(DatePart.minute, date, employee.datefield));
         int seconds = query.singleResult(SQLExpressions.datediff(DatePart.second, date, employee.datefield));
 
-        assertEquals(949449600, seconds);
-        assertEquals(15824160,  minutes);
-        assertEquals(263736,    hours);
-        assertEquals(10989,     days);
+        assertEquals(949363200, seconds);
+        assertEquals(15822720,  minutes);
+        assertEquals(263712,    hours);
+        assertEquals(10988,     days);
         assertEquals(361,       months);
         assertEquals(30,        years);
     }
@@ -471,7 +472,7 @@ public class SelectBase extends AbstractBaseTest {
     @Test
     public void DateTime() {
         TestQuery query = query().from(employee).orderBy(employee.id.asc());
-        assertEquals(Integer.valueOf(2),      query.singleResult(employee.datefield.dayOfMonth()));
+        assertEquals(Integer.valueOf(10),      query.singleResult(employee.datefield.dayOfMonth()));
         assertEquals(Integer.valueOf(2),      query.singleResult(employee.datefield.month()));
         assertEquals(Integer.valueOf(2000),   query.singleResult(employee.datefield.year()));
         assertEquals(Integer.valueOf(200002), query.singleResult(employee.datefield.yearMonth()));
@@ -1576,7 +1577,7 @@ public class SelectBase extends AbstractBaseTest {
     @ExcludeIn({DERBY, H2})
     public void YearWeek() {
         TestQuery query = query().from(employee).orderBy(employee.id.asc());
-        assertEquals(Integer.valueOf(200005), query.singleResult(employee.datefield.yearWeek()));
+        assertEquals(Integer.valueOf(200006), query.singleResult(employee.datefield.yearWeek()));
     }
 
 }

--- a/querydsl-sql/src/test/java/com/mysema/query/suites/H2SuiteTest.java
+++ b/querydsl-sql/src/test/java/com/mysema/query/suites/H2SuiteTest.java
@@ -1,7 +1,6 @@
 package com.mysema.query.suites;
 
 import com.mysema.query.*;
-import com.mysema.query.SpatialBase;
 import com.mysema.query.sql.spatial.GeoDBTemplates;
 import org.junit.BeforeClass;
 

